core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -159,10 +158,22 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
     }
 
     if (requests.isEmpty) {
-      // poll timeout of 0 doesn't work for the commits
-      val rawResult = consumer.poll(1)
-      if (rawResult.count > 0)
-        throw new IllegalStateException(s"Got ${rawResult.count} unexpected messages")
+
+      def checkNoResult(rawResult: ConsumerRecords[K, V]): Unit =
+        if (!rawResult.isEmpty)
+          throw new IllegalStateException(s"Got ${rawResult.count} unexpected messages")
+
+      checkNoResult(consumer.poll(0))
+
+      var i = 10
+      while (i > 0 && commitsInProgress > 0) {
```

Comment:
commitAsync takes around 200 us on my machine, so poll(1) is wasteful because it means blocking for 1500 us. Therefore we first try poll(0) in a "busy loop". On my machine that succeeds after around 5th attempt with the current 10 us parkNanos.